mmv1/products/compute/NetworkEndpoints.yaml

```diff
@@ -0,0 +1,139 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'NetworkEndpoints'
+kind: 'compute#networkEndpoints'
+base_url: 'projects/{{project}}/zones/{{zone}}/networkEndpointGroups/{{network_endpoint_group}}'
+description: |
+  A set of network endpoints belonging to a network endpoint group (NEG). A
+  single network endpoint represents a IP address and port combination that is
+  part of a specific (NEG). NEGs are zonal collections of these endpoints for
+  GCP resources within a single subnet. **NOTE**: Network endpoints cannot be
+  created outside of a network endpoint group.
+
+  This resource is authoritative for a single NEG. Any endpoints not specified
+  by this resource will be deleted when the resource configuration is applied.
+immutable: true
```

Comment:
I think this resource actually shouldn't be immutable, since the user can remove some network endpoints without removing all of them. That is, right now removing one network endpoint from this resource will delete and recreate all of them.

If you make this resource updatable, you can/should make the tests fail if the resource is deleted as described in https://github.com/hashicorp/terraform-provider-google/issues/12594 There are a few different existing patterns to get the lifecycle block in place - here's an example: https://github.com/hashicorp/terraform-provider-google/blob/9399d39ad2aab46ea6e6e612165eb48ecb5171fd/google/resource_redis_instance_test.go#L115

Reply:
In terms of getting the update to work properly - I _think_ you could probably do something like reusing the create_verb and create_url as update_verb and update_url to generate the update code and use a custom_code.update_encoder to fetch the list of remote endpoints, delete any that are not in terraform, and set the "update" list to just be the ones that don't exist in the API yet.

Reply:
I think to properly make this mutable, we need to be able to fetch all the endpoints... which means adding pagination. Incidentally I don't think this can be authoritative without pagination since the resource doesn't know anything about the 501st+ endpoint.

I took a stab at implementing pagination and it seems to work nicely, lmk what you think.

Reply:
The pagination implementation looks great, thanks for adding it! Makes sense re: needing it in order to be authoritative as well.